homeassistant/components/recorder/purge.py

```diff
@@ -248,26 +332,52 @@ def _purge_filtered_data(instance: Recorder, session: Session) -> bool:
     return True
 
 
+def _remove_attributes_ids_used_by_other_entities(
+    session: Session, entities: list[str], attributes_ids: set[int]
+) -> set[int]:
+    """Remove attributes ids that are still in use for entitiy_ids we are not purging yet."""
+    if not attributes_ids:
+        return set()
```

Comment:
Can we cover this in a test?

Reply:
762c015fea38049c53da437038ccf48c60a0319c

Reply:
Verified cover